src/stories/Forms/Select/Select.tsx

```diff
@@ -0,0 +1,489 @@
+    throw new Error('Select component: children can only contain Option components.');
+  }
+
+  const selectName = name || id;
+  const placeholder = '- Select option -';
+
+  const selectWrapperRef = useRef<HTMLDivElement>(null);
+  const selectButtonRef = useRef<HTMLButtonElement>(null);
+  const selectListRef = useRef<HTMLUListElement>(null);
+  const selectListHoverBackgroundRef = useRef<HTMLSpanElement>(null);
+  const filterSearchRef = useRef<HTMLInputElement>(null);
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [selected, setSelected] = useState<string | null>(value);
+  const [selectListHeight, setSelectListHeight] = useState(0);
+  const [filterSearch, setFilterSearch] = useState('');
+
+  useEffect(() => {
+    setSelected(value);
+  }, [value]);
+
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      const selectList = selectListRef.current;
+      const selectListChildren = selectList?.children as HTMLCollectionOf<HTMLElement>;
+      if (!selectListChildren) return;
+
+      const activeElement = document.activeElement as HTMLElement;
+      const activeElementIndex = Array.from(selectListChildren).indexOf(activeElement);
+
+      if (['Escape', 'ArrowDown', 'ArrowUp', 'Enter', ' '].includes(e.key)) {
+        e.preventDefault();
+      }
+
+      switch (e.key) {
+        case ' ':
+          if (activeElement === selectButtonRef.current && !isOpen) {
+            setIsOpen(true);
+          }
+          break;
+
+        case 'Escape':
+          setIsOpen(false);
+          selectButtonRef?.current?.focus();
+          break;
+
+        case 'ArrowUp':
+          if (activeElementIndex === -1) return;
+          if (activeElementIndex === 0) {
+            selectList.focus();
+          } else {
+            selectListChildren[activeElementIndex - 1].focus();
+          }
+          break;
+
+        case 'ArrowDown':
+          if (activeElementIndex === -1) return;
+          if (activeElementIndex === selectListChildren.length - 1) {
+            selectList.focus();
+          } else {
+            selectListChildren[activeElementIndex + 1].focus();
+          }
+          break;
+
+        case 'Enter':
+          if (activeElementIndex !== -1) {
+            selectListChildren[activeElementIndex].click();
+          }
+          break;
+        case 'Tab':
+          filterSearchRef?.current?.setAttribute('tabindex', activeElement === selectButtonRef.current ? '-1' : '0');
+          break;
+        default:
+          if (!/^[a-zA-Z0-9]$/.test(e.key) || !isOpen) return;
+          if (filterSearchRef.current) {
+            filterSearchRef.current.focus();
+          }
+          break;
+      }
+    };
+
+    window.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [isOpen]);
+
+  useEffect(() => {
+    setSelectListHeight(getSelectListHeight(showItems, selectListRef));
```

Comment:
Could maybe setSelecListHeight only be done on the initial render?

Reply:
It has to change depending on the filter result and if children change dynamically.